test/units/module_utils/common/test_collections.py

```diff
@@ -126,13 +127,13 @@ def test_from_kwargs(self):
     def test_immutable(self):
         imdict = ImmutableDict({1: 2})
 
-        with pytest.raises(TypeError) as exc_info:
+        expected_reason = r"^'ImmutableDict' object does not support item assignment$"
+
+        with pytest.raises(TypeError, match=expected_reason):
```

Comment:
Be sure that this fails in shippable on error.  For a while we were pinned on a version of pytest where match did not work.

Reply:
Well, master seems green + I've checked it locally in docker containers. I assume that pytest should be the same...

Reply:
I mean, you need to test that it properly fails if you raise the correct exception but with the wrong message.  The last time I checked in shippable (but could have been as much as a year ago) pytest's match parameter didn't work.  If I raised ```TypeError('Incorrect message')``` pytest.raises would have thought it was fine.

Reply:
(It correctly fails locally, so I am pretty sure that the latest pytest doesn't have the problem.  But I don't know if we're still pinning an older version of pytest in shippable which will not handle match right)

Reply:
It's `pytest-4.4.0` according to https://app.shippable.com/github/ansible/ansible/runs/125466/10/console